sdk/src/main/java/com/google/cloud/dataflow/sdk/io/datastore/V1Beta3.java

```diff
@@ -747,48 +859,40 @@ public void startBundle(Context c) {
 
     @Override
     public void processElement(ProcessContext c) throws Exception {
-      // Verify that the entity to write has a complete key.
-      if (!isValidKey(c.element().getKey())) {
-        throw new IllegalArgumentException(
-            "Entities to be written to the Datastore must have complete keys");
-      }
-
-      entities.add(c.element());
-      if (entities.size() >= V1Beta3.DATASTORE_BATCH_UPDATE_LIMIT) {
+      mutations.add(c.element());
+      if (mutations.size() >= V1Beta3.DATASTORE_BATCH_UPDATE_LIMIT) {
         flushBatch();
       }
     }
 
     @Override
     public void finishBundle(Context c) throws Exception {
-      if (entities.size() > 0) {
+      if (mutations.size() > 0) {
```

Comment:
missed this earlier -- this should just be `!mutations.isEmpty()`

Fix in followup.